### ISSUE TITLE
runtime/race: building for iOS, but linking in object file built for macOS

### ISSUE THREAD

**Issue by eliasnaur** [golang/go#46931]
### What version of Go are you using (`go version`)?

Happens on tip on the Corellium iOS builders that have recently upgraded to iOS 14.6 while debugging #35851.

### Does this issue reproduce with the latest release?

Yes

### What operating system and processor architecture are you using (`go env`)?

ios/arm64

### What did you do?

```
$ CC=$HOME/bin/clangwrap PATH=$PATH:$HOME/bin ./all.bash
```
### What did you expect to see?

No errors.

### What did you see instead?

```
# runtime/race.test
/var/root/goroot/pkg/tool/ios_arm64/link: running /var/root/bin/clangwrap failed: exit status 1
ld: building for iOS, but linking in object file built for macOS, file '/tmp/go-link-1066573738/000000.o' for architecture arm64
Not signing file
clang-5.0: error: linker command failed with exit code 1 (use -v to see invocation)

FAIL	runtime/race [build failed]
```

I'm guessing the problem is that the race runtime .syso files are hard-coded to macOS.

CC @cherrymui.

**Comment by cherrymui** [comment]
Yeah, the race detector is not implemented for iOS, only for macOS. Maybe we should skip that test.

**Comment by eliasnaur** [comment]
The test fails at build which is unfortunate. Perhaps the linker should skip .syso files whose LC_VERSION_MIN_*/LC_BUILD_VERSION load commands don't match GOOS?

**Comment by cherrymui** [comment]
@gopherbot please backport this to Go 1.16. This fixes a test currently failing on 1.16 branch.
